tests/src/python/test_qgssymbollayer_readsld.py

```diff
@@ -507,7 +507,7 @@ def test_read_circle(self):
         """Test wellknown name circle polygon fill"""
 
         sld = """<?xml version="1.0" encoding="UTF-8"?>
-            <StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ogc="http://www.opengis.net/ogc" version="1.1.0" xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.1.0/StyledLayerDescriptor.xsd" xmlns:se="http://www.opengis.net/se">
+            <StyledLayerDescriptor xmlns="http://www.opengis.net/sld" xmlns:sld="http://www.opengis.net/sld" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:ogc="http://www.opengis.net/ogc" version="1.1.0" xsi:schemaLocation="http://www.opengis.net/sld http://schemas.opengis.net/sld/1.1.0/StyledLayerDescriptor.xsd" xmlns:se="http://www.opengis.net/se">
```

Comment:
Doesn't this change what the test is checking? Presumably some tool output the original (invalid?) XML...

Reply:
Using a namespace without defining it with the xmlns: attribute is invalid XML. Cf
```
$ echo "<sld:foo/>" | xmllint --format -
-:1: namespace error : Namespace prefix sld on foo is not defined
<sld:foo/>
        ^
<?xml version="1.0"?>
<sld:foo/>
```
vs

```
$ echo "<sld:foo xmlns:sld='something'/>" | xmllint --format -
<?xml version="1.0"?>
<sld:foo xmlns:sld="something"/>
```

Reply:
Hmm - I'm still nervous that some tool out there is generating this content, and would lean towards just being tolerant and handling this automatically if we can.

Could we use a similar approach to https://github.com/qgis/QGIS/pull/61690/files#diff-2b31f4f76dacb4bd2240768e67fbe59d416219b9b58b3fece7706a1ebb18fbaeR3569 ?

Reply:
> Could we use a similar approach to

thanks for the hint! pull request updated accordingly